sys/include/event/thread.h

```diff
@@ -12,6 +12,39 @@
  * @ingroup     sys_event
  * @brief       Provides utility functions for event handler threads
  *
+ * Usage
+ * =====
+ *
+ * By using the module `event_thread`, the event queues
+ * @ref EVENT_PRIO_HIGHEST, @ref EVENT_PRIO_MEDIUM, and
+ * @ref EVENT_PRIO_LOWEST are provided and declared in the header
+ * `event/thread.h`. With default settings, one or more threads to handle these
+ * queues are automatically started during startup due to the module
+ * `auto_init`.
+ *
+ * By default, only a single thread with priority `EVENT_THREAD_MEDIUM_PRIO` is
+ * started that handles all three event queues according to their priority.
+ * An already started event handler will not be preempted in this case by an
+ * incoming higher priority event. Still, lower priority event queues will only
+ * be worked on once the higher priority queues are empty. Hence, the worst case
+ * latency is increased by the worst case duration of any possible lower
+ * priority event in this configuration.
+ *
+ * By using module `event_thread_highest`, the highest priority queue gets its
+ * own thread. With this, events of the highest priority can preempt already
+ * running event handlers of medium and lowest priority.
+ *
+ * By using module `event_thread_medium`, the lowest priority events are handled
+ * in their own thread. With this, events of at least medium priority can
+ * preempt already running events of the lowest priority.
+ *
+ * By using both module `event_thread_highest` and `event_thread_medium`, each
+ * event queue gets its own thread. So higher priority events will always
+ * preempt events of lower priority in this case.
+ *
+ * Finally, the module `event_thread_lowest` is provided for backward
+ * compatibility and has no effect.
```

Comment:
Why would this be needed? In case there are dependencies against this module?

Reply:
Exactly. We could deprecate it. But I think the configuration will change a bit once KConfig takes over. (I think with KConfig there will be better ways to configure the threads then using modules.)

Reply:
I think it makes sense to add a deprecation mention, and remove it from the test `Makefile`, we already have enough modules to add more that are unused.

Reply:
Unless I missed something, there was no infrastructure for module deprecation in place. I added this. It might make sense to split this out in to its own PR